synthesizer/snark/src/proving_key/mod.rs

```diff
@@ -20,7 +20,7 @@ mod serialize;
 
 use std::collections::BTreeMap;
 
-#[derive(Clone)]
+#[derive(Clone, PartialEq, Eq)]
```

Comment:
We have to be a little careful here, as well as in `CircuitProvingKey`; right now the `Eq` and `PartialEq` impls on `CircuitProvingKey` compare the *entire* struct, whereas the corresponding `Ord` impls compare only the circuit IDs. The latter is obviously much more efficient, and if the circuit ID is computed honestly, the two checks are essentially equivalent. However, we should be careful about not accidentally invoking equality checks on the entire struct, as that would be very expensive.

Reply:
Good point, made it less ambiguous: b6010842a